src/Compilers/CSharp/Test/Semantic/Semantics/LambdaDiscardParametersTests.cs

```diff
@@ -0,0 +1,542 @@
+                return 2;
+            };
+    }
+}");
+
+            comp.VerifyDiagnostics(
+                // (9,17): error CS0177: The out parameter '_' must be assigned to before control leaves the current method
+                //                 return 2;
+                Diagnostic(ErrorCode.ERR_ParamUnassigned, "return 2;").WithArguments("_").WithLocation(9, 17)
+                );
+        }
+
+        [Fact]
+        public void DiscardParameters_OnLocalFunction()
+        {
+            var comp = CreateCompilation(@"
+class C
+{
+    static void M()
+    {
+        local(1, 2);
+        void local(int _, int _) { }
+    }
+}");
+
+            comp.VerifyDiagnostics(
+                // (7,31): error CS0100: The parameter name '_' is a duplicate
+                //         void local(int _, int _) { }
+                Diagnostic(ErrorCode.ERR_DuplicateParamName, "_").WithArguments("_").WithLocation(7, 31)
+                );
+        }
+
+        [Fact]
+        public void DiscardParameters_UnicodeUnderscore()
+        {
+            var comp = CreateCompilation(@"
+public class C
+{
+    public static void Main()
+    {
+        System.Func<short, short, long> f1 = (\u005f, \u005f) => 3L;
+        \u005f = 1;
+    }
+}");
+            comp.VerifyDiagnostics(
+                // (6,55): error CS0100: The parameter name '_' is a duplicate
+                //         System.Func<short, short, long> f1 = (\u005f, \u005f) => 3L;
+                Diagnostic(ErrorCode.ERR_DuplicateParamName, @"\u005f").WithArguments("_").WithLocation(6, 55),
+                // (7,9): error CS0103: The name '_' does not exist in the current context
+                //         \u005f = 1;
+                Diagnostic(ErrorCode.ERR_NameNotInContext, @"\u005f").WithArguments("_").WithLocation(7, 9)
+                );
+        }
+
+        [Fact]
+        public void DiscardParameters_EscapedUnderscore()
+        {
+            var comp = CreateCompilation(@"
+public class C
+{
+    public static void Main()
+    {
+        System.Func<short, short, long> f1 = (@_, @_) => 3L;
+        @_ = 1;
+    }
+}");
+            comp.VerifyDiagnostics(
+                // (6,51): error CS0100: The parameter name '_' is a duplicate
+                //         System.Func<short, short, long> f1 = (@_, @_) => 3L;
+                Diagnostic(ErrorCode.ERR_DuplicateParamName, "@_").WithArguments("_").WithLocation(6, 51),
+                // (7,9): error CS0103: The name '_' does not exist in the current context
+                //         @_ = 1;
+                Diagnostic(ErrorCode.ERR_NameNotInContext, "@_").WithArguments("_").WithLocation(7, 9)
+                );
+        }
+
+        [Fact]
+        public void DiscardParameters_SingleUnderscoreParameter()
+        {
+            var comp = CreateCompilation(@"
+public class C
+{
+    public static void Main()
+    {
+        System.Func<short, short, long> f1 = (_, a) =>
+        {
+            int _ = 0; // 1
```

Comment:
Probably good idea to test for no regression on C# 8.0 shadowing with underscore?
```cs
        Action<int> f1 = (_) =>
        {
             _.ToString(); // ok
            Action<int> g2 = (_) => _.ToString(); // ok
        };
        Action<int, int> f2 = (_, _) =>
        {
             _.ToString(); // error
            Action<int> g2 = (_) => _.ToString(); // ok
        };
```

Reply:
Added to test plan. Thanks

Reply:
Added ` DiscardParameters_Shadowing`